src/index.rs

```diff
@@ -795,6 +796,45 @@ impl Index {
       .collect()
   }
 
+  pub(crate) fn get_children_by_inscription_id_with_limit(
```

Comment:
Let's merge these two functions. We can add a page size argument to `get_children_by_inscription_id_paginated`, and request the first page with page size 4.